src/Utilities/Functional.hpp

```diff
@@ -195,6 +223,11 @@ struct Identity : Functional<1> {
   }
 };
 
+template <int val>
+struct Literal : Functional<0> {
```

Comment:
What are the use cases for this that you're imagining? All this seems to do is replace an argument in `(...)` with one in the template list. (I'm not saying we don't want it, it's just not clear from this PR what the benefit of splitting numbers into two places is)

Reply:
The documentation I've added at the start of the namespace tries to clarify that - the usecase that I'm imagining is trying to assemble, for instance, a set of types which represent useful compile-time functions. For instance, say you want to create a set of one-dimensional probability density functions, which you'll want to all be Functional<1>'s. So, you want the ability to represent the full chain of ratios, products, etc that go on with compile-time literals in place. Once assembled, they could then be called on a template struct that does something with probability functions, like computes statistics without needing to know all the messy parameters.

Reply:
Okay, I can accept that :)